Tasks/JavaToolInstaller/FileExtractor/JavaFilesExtractor.ts

```diff
@@ -0,0 +1,145 @@
+import path = require('path');
+
+export class JavaFilesExtractor {
+    public destinationFolder: string;
+    public win: RegExpMatchArray;
+
+    // 7zip
+    public xpSevenZipLocation: string;
+    public winSevenZipLocation: string = path.join(__dirname, '7zip/7z.exe');
+
+    constructor() {
+        this.win = taskLib.osType().match(/^Win/);
+        taskLib.debug('win: ' + this.win);
+    }
+
+    private getSevenZipLocation(): string {
+        if (this.win) {
+            return this.winSevenZipLocation;
+        } else {
+            if (typeof this.xpSevenZipLocation == "undefined") {
+                this.xpSevenZipLocation = taskLib.which('7z', true);
+            }
+            return this.xpSevenZipLocation;
+        }
+    }
+
+    private async unzipExtract(file, destinationFolder) {
+        console.log(taskLib.loc('UnzipExtractFile', file));
+        if (process.platform == 'win32') {
+            // build the powershell command
+            let escapedFile = file.replace(/'/g, "''").replace(/"|\n|\r/g, ''); // double-up single quotes, remove double quotes and newlines
+            let escapedDest = destinationFolder.replace(/'/g, "''").replace(/"|\n|\r/g, '');
+            let command: string = `$ErrorActionPreference = 'Stop' ; try { Add-Type -AssemblyName System.IO.Compression.FileSystem } catch { } ; [System.IO.Compression.ZipFile]::ExtractToDirectory('${escapedFile}', '${escapedDest}')`;
+    
+            // change the console output code page to UTF-8.
+            let chcpPath = path.join(process.env.windir, "system32", "chcp.com");
+            await taskLib.exec(chcpPath, '65001');
+    
+            // run powershell
+            let powershell: tr.ToolRunner = taskLib.tool('powershell')
+                .line('-NoLogo -Sta -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command')
+                .arg(command);
+            await powershell.exec();
+        }
+        else {
+            let unzip: tr.ToolRunner = taskLib.tool('unzip')
+                .arg(file);
+            await unzip.exec(<tr.IExecOptions>{ cwd: destinationFolder });
+        }
+    }
+
+    private async sevenZipExtract(file, destinationFolder) {
+        console.log(taskLib.loc('SevenZipExtractFile', file));
+        var sevenZip = taskLib.tool(this.getSevenZipLocation());
+        sevenZip.arg('x');
+        sevenZip.arg('-o' + destinationFolder);
+        sevenZip.arg(file);
+        var execResult = sevenZip.execSync();
+        if (execResult.code != taskLib.TaskResult.Succeeded) {
+            taskLib.debug('execResult: ' + JSON.stringify(execResult));
+        }
+    }
+
+    private async tarExtract(file, destinationFolder) {
+        console.log(taskLib.loc('TarExtractFile', file));
+        let tr: tr.ToolRunner = taskLib.tool('tar');
+        tr.arg(['xzC', destinationFolder, '-f', file]);
+        tr.exec();
+    }
+
+    private extractFiles(file: string, fileEnding: string) {
+        var stats = taskLib.stats(file);
```

Comment:
search and replace var. I would try replacing it with const and then fixing all the places where it has to be a let.